dropshot/src/handler.rs

```diff
@@ -486,9 +512,18 @@ where
          * resolved statically.
          */
         let rqctx = Arc::new(rqctx_raw);
+        let response_rqctx = rqctx.clone();
         let funcparams = Extractor::from_request(Arc::clone(&rqctx)).await?;
         let future = self.handler.handle_request(rqctx, funcparams);
-        future.await
+
+        future.await.map(|mut response| {
+            let mutable_state = response_rqctx.mutable_state.lock().unwrap();
+
+            for (header, value) in &mutable_state.response_headers {
+                response.headers_mut().insert(header, value.clone());
```

Comment:
As written, it looks like if somebody sets the value multiple times with `add_response_header`, we'll store them all in the Vec but the end result will be to only take the last one.

Reply:
That's as I intended. What do you think would be a better interface?

Reply:
I think you resolved this separately by using the HeaderMap.